### ISSUE TITLE
More descriptive error when server_error encountered

### ISSUE THREAD

**Issue by sagor999** [jacksontj/promxy#301]
Would it be possible to provide a bit more information when this error occurs:
`promxy {"caller":"manager.go:512","err":"server_error: server error: 502","group":"/rules/synology.rules","level":"warn","msg":"Evaluating rule failed"," rule":"<rule description skipped>"} `

It would be very helpful it in that error it would say which server group and which specific prometheus returned that error.

**Comment by jacksontj** [comment]
Thanks for reaching out! As some background context this error is actually being logged out by the prometheus rule manager (https://github.com/prometheus/prometheus/blob/fc3fb3265a0f0061d07d0bf5fb4103e2144565b2/rules/manager.go#L577) -- so we have limited control over the message itself. But from the code it does lot the `err` as it sees it, so we have some control.

I did a bit of looking into this, and its potentially doable-- but may end up being difficult. I try to use as much upstream prom code as possible (to reduce the overhead in creating and maintaining this project) so that error is actually coming from the promethus API client. I have additional logic internally that wraps errors with additional context, but the prometheus package doesn't unwrap those -- so I have to before returning them back (e.g. https://github.com/jacksontj/promxy/blob/master/pkg/proxyquerier/querier.go#L58).

So a couple ways I see to addressing this:
- have the prometheus API client add the target to its log message
- stop doing `.Cause()` on errors before handing to prometheus (pretty sure this will break error handling)
- add support for prometheus to call `Cause()` itself


The first one seems the easiest, although I'm not sure that they'd be open to such a change. The other 2 are both likely massive changes. 


That all being said, we could at most easily add *which* server it was trying to talk to -- "which servergroup" is a bit harder since servergroups don't have names etc (best we could hack up is an ordinal).
